packages/cli/src/lib/decorators/preconditions/preconditions.ts

```diff
@@ -3,7 +3,7 @@ import type Command from '@oclif/command';
 export type PreconditionFunction = (
   target: Command,
   instance?: Command
-) => never | void | Promise<never | void>;
+) => Promise<never | void>;
 
```

Comment:
I think it does not make sense to not return a promise if at the end we do
```
await precondition.call(this, target);
```

Reply:
> I think it does not make sense to not return a promise if at the end we do
> 
> ```
> await precondition.call(this, target);
> ```

An async function always returns a Promise.

Reply:
yes, that's what I am saying :) 
hence the `never | void` removal

Reply:
My bad! I misread your comment at first lgtm! 👍